RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -0,0 +1,1481 @@
+        auto soiSamples = cms::alpakatools::make_device_buffer<int8_t[]>(queue, totalChannels * windowSize);
+
+        alpaka::exec<Acc2D>(queue,
+                            workDivPrep2D,
+                            mahi::kernel_prep1d_sameNumberOfSamples{},
+                            outputGPU,
+                            f01HEDigis,
+                            f5HBDigis,
+                            f3HBDigis,
+                            mahi,
+                            sipmCharacteristics,
+                            recoParamsWithPS,
+                            configParameters.useEffectivePedestals,
+                            configParameters.sipmQTSShift,
+                            configParameters.sipmQNTStoSum,
+                            configParameters.firstSampleShift,
+                            configParameters.ts4Thresh,
+                            reinterpret_cast<float*>(amplitudes.data()),
+                            reinterpret_cast<float*>(noiseTerms.data()),
+                            reinterpret_cast<float*>(electronicNoiseTerms.data()),
+                            reinterpret_cast<float*>(soiSamples.data()),
+                            windowSize);
+
+        //// 1024 is the max threads per block for gtx1080
+        //// FIXME: Take this from Alpaka in a way that does not need to query deviceProperty for every event
+        uint32_t const channelsPerBlock = 1024 / (windowSize * mahiPulseOffsets.metadata().size());
+
+        //launch 1D blocks of 3D threads
+        auto const blocks_z = cms::alpakatools::divide_up_by(totalChannels, channelsPerBlock);
+        Vec3D const blocks_3d{blocks_z, 1u, 1u};  // 1D block in z {z,y,x} coordinates
+        Vec3D const threads_3d{channelsPerBlock, mahiPulseOffsets.metadata().size(), windowSize};
+
+        auto workDivPrep3D = cms::alpakatools::make_workdiv<Acc3D>(blocks_3d, threads_3d);
+
+        //Device buffer for output
+        auto pulseMatrices = cms::alpakatools::make_device_buffer<float[]>(
+            queue, totalChannels * windowSize * mahiPulseOffsets.metadata().size());
+        auto pulseMatricesM = cms::alpakatools::make_device_buffer<float[]>(
+            queue, totalChannels * windowSize * mahiPulseOffsets.metadata().size());
+        auto pulseMatricesP = cms::alpakatools::make_device_buffer<float[]>(
+            queue, totalChannels * windowSize * mahiPulseOffsets.metadata().size());
+
+        alpaka::exec<Acc3D>(queue,
+                            workDivPrep3D,
+                            mahi::kernel_prep_pulseMatrices_sameNumberOfSamples{},
+                            reinterpret_cast<float*>(pulseMatrices.data()),
+                            reinterpret_cast<float*>(pulseMatricesM.data()),
+                            reinterpret_cast<float*>(pulseMatricesP.data()),
+                            mahiPulseOffsets,
+                            reinterpret_cast<float*>(amplitudes.data()),
```

Comment:
This `reinterpret_cast` seems unnecessary

Reply:
removed